core/src/main/java/org/axonframework/config/AggregateConfigurer.java

```diff
@@ -82,13 +83,17 @@ protected AggregateConfigurer(Class<A> aggregate) {
                                  "Store to use, or configure a specific repository implementation for " +
                                  aggregate.toString());
             if (c.commandBus() instanceof DisruptorCommandBus) {
-                return ((DisruptorCommandBus) c.commandBus())
-                        .createRepository(c.eventStore(), aggregateFactory.get(),
-                                          snapshotTriggerDefinition.get(),
-                                          c.parameterResolverFactory());
+                return ((DisruptorCommandBus) c.commandBus()).createRepository(c.eventStore(),
+                                                                               aggregateFactory.get(),
+                                                                               snapshotTriggerDefinition.get(),
+                                                                               c.parameterResolverFactory(),
+                                                                               c::repository);
```

Comment:
Ah, so this to me suggests the default `RepositoryProvider` will only provide the aggregate being configured, so not any potential other Aggregates that Aggregate might create. Is that a correct assumption?

Reply:
And, if this is the default from Axon's configuration, why ever make it null? Is that to not break users which will move from an earlier Axon version to 3.3?

Reply:
Setting `null` is for backward compatibility reasons.

Reply:
Thought as much, thanks for confirming that.